### PR TITLE
Http call into nstableview

### PR DESCRIPTION
**Pull request by KenGorab**

### CODE REVIEW

File: FileUpload.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -18,6 +18,7 @@
 		169CBE621AE3812C00D237C7 /* FileModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 169CBE611AE3812C00D237C7 /* FileModel.swift */; };
```
Comment (KenGorab) [inline FileUpload.xcodeproj/project.pbxproj]: I'll bet this file could be added to the .gitignore too

**Comment by franktisellano** [comment]
WHAAAT YOU'RE AWESOME

File: FileUpload/Base.lproj/Main.storyboard [hunk 2]
```diff
@@ -688,7 +688,7 @@
                                     <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                                     <subviews>
                                         <tableView verticalHuggingPriority="750" allowsExpansionToolTips="YES" columnAutoresizingStyle="lastColumnOnly" alternatingRowBackgroundColors="YES" columnReordering="NO" columnResizing="NO" multipleSelection="NO" emptySelection="NO" autosaveColumns="NO" rowSizeStyle="automatic" viewBased="YES" id="4cJ-Qr-cBP">
-                                            <rect key="frame" x="0.0" y="0.0" width="238" height="117"/>
+                                            <rect key="frame" x="0.0" y="0.0" width="350" height="0.0"/>
```
Comment (KenGorab) [inline FileUpload/Base.lproj/Main.storyboard]: I don't recall changing this...

File: FileUpload/FileModel.swift [hunk 3]
```diff
@@ -9,22 +9,16 @@
 import Cocoa
 
 struct FileModel {
-    var id: Int?
-    var original_filename: String?
-    var aws_url: String?
-//    var uuid: String?
-    var slug: String?
-    var url: String?
+    let id: Int?
```
Comment (KenGorab) [inline FileUpload/FileModel.swift]: I prefer my objects to be immutable.

File: FileUpload/ViewController.swift [hunk 4]
```diff
@@ -14,67 +14,28 @@ class ViewController: NSViewController, NSTableViewDelegate, NSTableViewDataSour
...
-    
-    @IBAction func refreshData(sender: AnyObject) {
-        dataFromServer()
-    }
-
-    @IBAction func toggleMenu(sender: AnyObject) {
-        
-    }
-    
-    func dataFromServer() {
```
Comment (KenGorab) [inline FileUpload/ViewController.swift]: This has migrated to FileService.swift

File: Pods/Pods.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -7,266 +7,266 @@
 	objects = {
```
Comment (KenGorab) [inline Pods/Pods.xcodeproj/project.pbxproj]: This could also go in the .gitignore

File: FileUpload/ViewController.swift [hunk 6]
```diff
@@ -14,67 +14,28 @@ class ViewController: NSViewController, NSTableViewDelegate, NSTableViewDataSour
...
         })
     }
-
-}
-
-extension ViewController: NSTableViewDataSource, NSTableViewDelegate {
     
     func numberOfRowsInTableView(tableView: NSTableView) -> Int {
-        return files!.count
+        switch (self.files) {
```
Comment (KenGorab) [inline FileUpload/ViewController.swift]: I'm not a **huge** fan of the way swift handles pattern matching

**Pull request merged**